rotkehlchen/externalapis/etherscan_like.py

```diff
@@ -845,3 +847,74 @@ def eth_call(
             method='eth_call',
             options={'to': to_address, 'data': input_data},
         )
+
+    def get_logs(
+            self,
+            chain_id: SUPPORTED_CHAIN_IDS,
+            contract_address: ChecksumEvmAddress,
+            topics: list[str],
+            from_block: int,
+            to_block: int | str = 'latest',
+            existing_events: list[dict[str, Any]] | None = None,
+    ) -> list[dict[str, Any]]:
+        """Get logs for a given contract address and topics.
+        May raise RemoteError if there are problems contacting the indexer or the response has
+        invalid data.
+        """
+        options = {'fromBlock': from_block, 'toBlock': to_block, 'address': contract_address}
+        for idx, topic in enumerate(topics):
+            if topic is not None:
+                options[f'topic{idx}'] = topic
+                if idx == 0:
+                    continue
+
+                for i in range(idx):  # Specify the operator for each topic combination
+                    options[f'topic{i}_{idx}_opr'] = 'and'
```

Comment:
Something like this is more clear I believe


```suggestion
for idx, topic in enumerate(topics):
    if topic is None:
        continue

    options[f"topic{idx}"] = topic

    # Add AND-operators for all topic combinations with this index
    if idx > 0:
        for j in range(idx):
            options[f"topic{j}_{idx}_opr"] = "and"
```